File: evd_ros_backend/evd_ros_core/src/evd_script/program/tasks/open_gripper.py
from ..task import Task
from ..primitives import Gripper


class OpenGripper(Task):

    '''
    Data structure methods
    '''

    @classmethod
    def type_string(cls):
        return 'open-gripper.'

    @classmethod
    def full_type_string(cls):
        return Task.full_type_string() + cls.type_string()

    def __init__(self, position=0, effort=100, speed=100, thing_uuid=None, type='',
                 name='', uuid=None, parent=None, append_type=True, primitives=None):

        if primitives == None:
            primitives=[
                Gripper(
                    thing_uuid=thing_uuid,
                    position=position,
                    effort=effort,
                    speed=speed)
            ]

        super(OpenGripper,self).__init__(
            type='open-gripper.'+type if append_type else type,
            name=name,
            uuid=uuid,
            parent=parent,
            append_type=append_type,
            primitives=primitives)

File: evd_authoring_app/src/model/evdScript/program/skills/openGripper.js
import { Skill } from '../skill';
import { Gripper } from '../primitives';
import { NodeParser } from '../../utilityFunctions';
import Blockly from 'blockly';


export class OpenGripper extends Skill {

    /*
    * Data structure methods
    */

    static typeString() {
        return 'open-gripper';
    }

    static fullTypeString() {
        return Skill.fullTypeString() + OpenGripper.typeString();
    }

    constructor(thingUuid=null, position=0, effort=100, speed=100, type='', name='', 
                uuid=null, parent=null, appendType=true, primitives=null) 
    {

        if (primitives !== null) {
            primitives = [
                Gripper(
                    thingUuid,
                    position,
                    effort,
                    speed
                )
            ];
        }

        super(
            primitives,
            (appendType) ? 'open-gripper.'+type : type,
            name,
            uuid,
            parent,
            appendType
        );
    }

    static fromDict(dct) {
        return new OpenGripper(
            null,
            0,
            100,
            100,
            dct.type,
            dct.name,
            dct.uuid,
            null,
            false,
            dct.primitives.map(p => NodeParser(p))
        );
    }

    static BlocklyToolbox() {
        return { type: 'open_gripper' };
    }

    static BlocklyBlock() {
        return { key: 'open_gripper', data: {
            init: function() {
                this.appendDummyInput()
                    .appendField("Open Gripper");
                this.appendValueInput("thing")
                    .setCheck("thing")
                    .appendField("Thing");
                this.appendDummyInput()
                    .appendField("Position")
                    .appendField(new Blockly.FieldNumber(0, 0, 0), "position")
                    .appendField("%");
                this.appendDummyInput()
                    .appendField("Effort")
                    .appendField(new Blockly.FieldNumber(100, 0, 100), "effort")
                    .appendField("%");
                this.appendDummyInput()
                    .appendField("Speed")
                    .appendField(new Blockly.FieldNumber(100, 0, 100), "speed")
                    .appendField("%");
                this.setInputsInline(false);
                this.setPreviousStatement(true, null);
                this.setNextStatement(true, null);
                this.setColour(210);
                this.setTooltip("open-gripper");
                this.setHelpUrl("open-gripper");
            }
        }};
    }
}